File: week_2/pset_2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);                              //Count the number of letters
int count_words(string text);                                //Count the number of words
int count_sentences(string text);                            //Count the number of sentences

int main(void)
{
    string text = get_string("Introduce the text: ");         // Prompt the user for the text

    float L = (count_letters(text) / (float) count_words(text)) * 100;    // Avg number of letters per 100 words in the text
    float S = (count_sentences(text) / (float) count_words(text)) * 100;  // Avg number of sentences per 100 words in the text
    
    float index = (0.0588 * L) - (0.296 * S) - 15.8;          // Coleman-Liau Index

    if (index < 1)                                            // Printing the Grade
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        text[i] = tolower(text[i]);
        if (isalpha(text[i]))
        {
            count += 1;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        text[i] = tolower(text[i]);
        if (isspace(text[i]))
        {
            count += 1;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        text[i] = tolower(text[i]);
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            count += 1;
        }
    }
    return count;
}

File: week_6/pset_6/sentimental-readability/readability.py
from cs50 import get_string


def count_letters(string):
    # Count the number of letters
    count = 0
    string = string.lower()

    for i in range(len(string)):
        if string[i].isalpha():
            count += 1

    return count


def count_words(string):
    # Count the number of words
    count = 1
    string = string.lower()

    for i in range(len(string)):
        if string[i].isspace():
            count += 1

    return count


def count_sentences(string):
    # Count the number of sentences
    count = 0
    string = string.lower()

    for i in range(len(string)):
        if (string[i] == '.' or string[i] == '?' or string[i] == '!'):
            count += 1

    return count


def main():
    string = get_string("Introduce the text: ")         # Prompt the user for the text

    L = (count_letters(string) / count_words(string)) * 100     # Avg number of letters per 100 words in the text
    S = (count_sentences(string) / count_words(string)) * 100   # Avg number of sentences per 100 words in the text

    index = (0.0588 * L) - (0.296 * S) - 15.8          # Coleman-Liau Index

    if index < 1:                                            # Printing the Grade
        print("Before Grade 1")
    elif (index >= 16):
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")


if __name__ == '__main__':
    main()